### ISSUE TITLE
[Enhancement]: OpenID Connect add setting for requested scopes

### ISSUE THREAD

**Issue by exu-g** [advplyr/audiobookshelf#3006]
### Type of Enhancement

Server Backend

### Describe the Feature/Enhancement

Please add a setting to request specific scopes in addition to the 3 basic ones.  

Something similar to this, as offered by the OpenID Connect addon for Nextcloud.

![image](https://github.com/advplyr/audiobookshelf/assets/61015582/0c86ded1-be18-4a8f-8161-2dcecf869ca5)

### Why would this be helpful?

In practice, it's possible to have different names for the scope and its contained claim.  
I usually use the application name for the scope and have a different name for the actual claim for consistency on the application side.    

Scope:
![image](https://github.com/advplyr/audiobookshelf/assets/61015582/8b080734-df05-46cd-8dd8-cb83590cc812)

Claim:
![image](https://github.com/advplyr/audiobookshelf/assets/61015582/3bda4312-460f-4fd7-8b8d-66576f1b1244)

With the current settings this does not work, either I set the correct claim and the scope won't be sent at all, or I use the working scope, but can't find any groups within.

### Future Implementation (Screenshot)

A simple field titled "additional scopes" would be fine.  

### Audiobookshelf Server Version

v2.9.0

### Current Implementation (Screenshot)

Settings > Authentication > OpenID Connect Authentication
![image](https://github.com/advplyr/audiobookshelf/assets/61015582/922465a1-b848-4ab2-9d5c-d3ebe690eef0)

**Comment by Sapd** [comment]
> I guess adding a note that claim == scope name in the documentation would suffice, so others don't have to spend time wondering why the heck stuff doesn't work.

Yeah I thought (when I developed that feature) to show the claims as greyed-out box which would update with the name of the group claims etc., which would make claim == scope then clear.

But you also got a point that it makes sense to add a box and to allow to add claims to allow configurations as you showed. There then just must be a warning to the user that he also has to put in the name of the scope which contains the claim explicitly in the additional claims field.

PS: I know you just made example pictures, but to prevent you a debugging session, keep in mind that in "abspermissions" in Authentik you have to write the booleans als true python booleans like this. Otherwise they won't be provided as JSON boolean types:
```python
abspermissions = {
  "abspermissions": {
    "canDownload": False, # Upper case without "
     "canUpload": True
     ....
   }
}
```

**Comment by igbjnI05bF** [comment]
I'm very confused here, most of this is above my head. I'm unable to get the group claim to work at all with keycloak. I've done this successfully with other apps such as Nextcloud, Jellyfin, and Grafana, but cannot get it to work with Audiobookshelf.

In my Audiobookshelf client in Keycloak I went to client scopes, audiobookshelf-dedicated, and added a client roles scope. I named the token claim name "absroles". I then went to the roles tab of audiobookshelf and created "admin", "guest", and "user" roles. I added my user to the audiobookshelf/admin role.

I cannot sign into Audiobookshelf, it says "Error in callback". The webtools show "Invalid scopes: openid profile email absroles". I don't really get it, openid, profile, and email are global realm scopes. absroles isn't (and shouldn't) be a realm scope, it's a client scope.

Am I missing something here? How do I get Audiobookshelf to see this client scope I made and respect the client roles my users are in?

**Comment by megheaiulian** [comment]
I think the issue is at https://github.com/advplyr/audiobookshelf/blob/master/server/Auth.js#L514. 
It adds the `authOpenIDGroupClaim` as a scope value but I think it shouldn't  as that is just the property to make groups from.
